File: zig/armstrong-numbers/armstrong_numbers.zig
const std = @import("std");
const math = std.math;

fn countDigits(comptime IntType: type, num: IntType) u32 {
    var n: IntType = @abs(num);

    if (n == 0) return 1;

    var count: u32 = 0;
    while (n > 0) : (n /= 10)
        count += 1;

    return count;
}

pub fn isArmstrongNumber(num: u128) bool {
    const power = countDigits(u128, num);
    var n: u128 = num;
    var total: u128 = 0;

    while (n > 0) : (n /= 10)
        total += math.pow(u128, @rem(n, 10), power);

    return total == num;
}

File: python/armstrong-numbers/armstrong_numbers.py
from math import log10, floor


def is_armstrong_number(number: int) -> bool:
    if (number == 0):
        return True
    digit_count = floor(log10(number)) + 1
    total = sum(
        map(
            lambda i: ((number // 10**i) % 10) ** digit_count,
            range(digit_count)
        )
    )
    return total == number
